File: LeetCode/reverse_integer.py
"""
Given a 32-bit signed integer, reverse digits of an integer.

Example 1:

Input: 123
Output: 321
Example 2:

Input: -123
Output: -321
Example 3:

Input: 120
Output: 21
"""


class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        reversed_number = 0
        while x > 0:
            reversed_number = reversed_number * 10
            reversed_number = reversed_number + x % 10;
            x = x / 10
        return reversed_number


if __name__ == "__main__":
    input = 123
    print Solution().reverse(input)
    input = -123
    print Solution().reverse(input)
    input = 120
    print Solution().reverse(input)

File: LeetCode/reverse_integer.c

#include<stdio.h>
#include<stdlib.h>
#include<math.h>


int reverse(int x){
	int reversed_number = 0;
	if(x > pow(2, 31)-1 || x < -pow(2, 31)-1)
		return 0;
	while (x != 0){
		reversed_number = reversed_number * 10;
		reversed_number = reversed_number + x %10;
		x = x /10;
	}
	return reversed_number;
}


int main(){
	int a = 123;
	int output = reverse(a);
	printf("%d\n", output);
	int b = -123;
	int output_b = reverse(b);
	printf("%d\n", output_b);
	int c = 120;
	int output_c = reverse(c);
	printf("%d\n", output_c);
	int d = 1534236469;
	int output_d = reverse(d);
	printf("%d\n", output_d);
}